tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -811,10 +1073,10 @@ public static void test_hugeint() throws Exception {
 		assertEquals(rs.getLong("hi2"), -42L);
 		assertEquals(rs.getObject("hi3"), new BigInteger("454564646545646546545646545"));
 		assertEquals(rs.getObject("hi4"), new BigInteger("-454564646545646546545646545"));
-		assertEquals(rs.getBigDecimal("hi1"), new BigDecimal("42"));
-		assertEquals(rs.getBigDecimal("hi2"), new BigDecimal("-42"));
-		assertEquals(rs.getBigDecimal("hi3"), new BigDecimal("454564646545646546545646545"));
-		assertEquals(rs.getBigDecimal("hi4"), new BigDecimal("-454564646545646546545646545"));
+		// assertEquals(rs.getBigDecimal("hi1"), new BigDecimal("42"));
```

Comment:
Should this be commented out?

Reply:
I reworked the handling in the case of calling getBigDecimal on a non-decimal result. Now the tests are OK again, but this is on best effort basis and might break for other numbers.

In addition the tests now use the compareTo method, which ignores differences in scale as long as the numbers are identical. As we are heavily casting here, this seems to be appropriate.